sdks/java/extensions/python/src/test/java/org/apache/beam/sdk/extensions/python/PythonExternalTransformTest.java

```diff
@@ -184,6 +186,19 @@ public void generateArgsWithCustomType() {
     assertEquals(456, (int) receivedRow.getRow("field1").getInt32("intField"));
   }
 
+  @Test
+  public void generateArgsWithTypeHint() {
+    PythonExternalTransform<?, ?> transform =
+        PythonExternalTransform
+            .<PCollection<KV<String, String>>, PCollection<KV<String, Iterable<String>>>>from(
+                "DummyTransform")
+            .withArgs(PythonCallableSource.of("dummy data"))
+            .withTypeHint(
```

Comment:
Until BEAM-14458 is resolved, can we automatically infer at least this special case from within PythonExternalTransform?

Reply:
Done.